File: php/test/Creational/BuilderTest.php
<?

require_once 'src/Creational/Builder/BostonSoupBuffetBuilder.php';
require_once 'src/Creational/Builder/HonoluluSoupBuffetBuilder.php';

class Creational_BuilderTest extends PHPUnit_Framework_TestCase
{
	function createSoupBuffet($builder)
	{
        $builder->buildSoupBuffet();
        $builder->setSoupBuffetName();
        $builder->buildChickenSoup();
        $builder->buildClamChowder();
        $builder->buildFishChowder();    
        $builder->buildMinnestrone();
        $builder->buildPastaFazul();
        $builder->buildTofuSoup();
        $builder->buildVegetableSoup();
        return $builder->getSoupBuffet();
	}     
    

	function test()       
	{
		$bostonSoupBuffet = $this->createSoupBuffet(new BostonSoupBuffetBuilder());
		$this->assertEquals("At the " . $bostonSoupBuffet->soupBuffetName . $bostonSoupBuffet->getTodaysSoups(),
			"At the Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			'Testing Boston soup buffet');

		$honoluluSoupBuffet = $this->createSoupBuffet(new HonoluluSoupBuffetBuilder());
		$this->assertEquals("At the " .  $honoluluSoupBuffet->soupBuffetName .  $honoluluSoupBuffet->getTodaysSoups(),
			"At the Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			'Testing Honolulu soup buffet');

	}

}

File: go/creational/builder/builder_test.go
package builder



import (
	"testing"
	"../../lib/asserts"
	"../common"
)

func TestBuilder (t *testing.T) {

   createSoupBuffet := func( soupBuffetBuilder SoupBuffetBuilder) *common.SoupBuffet { 
        soupBuffetBuilder.BuildSoupBuffet()
        return soupBuffetBuilder.GetSoupBuffet()
   }     
   
   bostonSoupBuffet := createSoupBuffet( NewBostonSoupBuffetBuilder())
       asserts.Equals( t, "Boston soup buffet builder test",
			"At the Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + bostonSoupBuffet.GetSoupBuffetName() + bostonSoupBuffet.GetTodaysSoups())

   honoluluSoupBuffet := createSoupBuffet( NewHonoluluSoupBuffetBuilder())
       asserts.Equals( t, "Honolulu soup buffet builder test",
			"At the Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + honoluluSoupBuffet.GetSoupBuffetName() + honoluluSoupBuffet.GetTodaysSoups())
}      
